metricbeat/module/kubernetes/apiserver/metricset.go

```diff
@@ -54,20 +73,38 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch gathers information from the apiserver and reports events with this information.
 func (m *Metricset) Fetch(reporter mb.ReporterV2) error {
 	events, err := m.prometheusClient.GetProcessedMetrics(m.prometheusMappings)
-	if err != nil {
-		return fmt.Errorf("error getting metrics: %w", err)
-	}
 
-	for _, e := range events {
-		event := mb.TransformMapStrToEvent("kubernetes", e, nil)
-		if len(m.clusterMeta) != 0 {
-			event.RootFields.DeepUpdate(m.clusterMeta)
+	error_string := fmt.Sprintf("%s", err)
```

Comment:
`err.Error()` gives the error in string format.

Reply:
Reverting back to fmt as I am getting some weird errors when used Error

```
{"log.level":"error","@timestamp":"2024-12-16T12:33:46.124Z","log.origin":{"function":"runtime.panicmem","file.name":"runtime/panic.go","file.line":261},"message":"recovered from panic while fetching 'kubernetes/scheduler' for host 'localhost:10259'. Recovering, but please report this.","service.name":"metricbeat","error":{"message":"runtime error: invalid memory address or nil pointer dereference"},"stack":"github.com/elastic/elastic-agent-libs/logp.Recover\n\t/Users/andreasgkizas/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.17.4/logp/global.go:101\nruntime.gopanic\n\t/Users/andreasgkizas/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.9.darwin-arm64/src/runtime/panic.go:770\nruntime.panicmem\n\t/Users/andreasgkizas/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.9.darwin-arm64/src/runtime/panic.go:261\nruntime.sigpanic\n\t/Users/andreasgkizas/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.9.darwin-arm64/src/runtime/signal_unix.go:881\ngithub.com/elastic/beats/v7/metricbeat/module/kubernetes/scheduler.(*MetricSet).Fetch\n\t/Users/andreasgkizas/elastic/beats/metricbeat/module/kubernetes/scheduler/scheduler.go:126\ngithub.com/elastic/beats/v7/metricbeat/mb/module.(*metricSetWrapper).fetch\n\t/Users/andreasgkizas/elastic/beats/metricbeat/mb/module/wrapper.go:279\ngithub.com/elastic/beats/v7/metricbeat/mb/module.(*metricSetWrapper).startPeriodicFetching\n\t/Users/andreasgkizas/elastic/beats/metricbeat/mb/module/wrapper.go:251\ngithub.com/elastic/beats/v7/metricbeat/mb/module.(*metricSetWrapper).run\n\t/Users/andreasgkizas/elastic/beats/metricbeat/mb/module/wrapper.go:235\ngithub.com/elastic/beats/v7/metricbeat/mb/module.(*Wrapper).Start.func1\n\t/Users/andreasgkizas/elastic/beats/metricbeat/mb/module/wrapper.go:175","ecs.version":"1.6.0"}
```

Reply:
Ah, that's because err doesn't always exist, so you would have to verify it before using `err.Error()`.

Reply:
Fixed the errorString
I would go with fmt